src/functions.js

```diff
@@ -67,7 +80,103 @@ function get_click_handler( item ) {
 	}
 }
 
-export {
-	get_click_handler,
-	extend_toolbar,
+export function addProviderFilter() {
+	// Short circuit if we don't have providers
+	if ( ! AMF_DATA || ! AMF_DATA.hasOwnProperty( 'providers' ) ) {
+		return;
+	}
+
+	addInlineStyle( `
+		.amf-hidden { display: none !important; }
+	` );
+
+	// If we have only 1 filter then it's the default, no need for a filter.
+	if ( Object.keys( AMF_DATA.providers ).length < 2 ) {
+		const provider = Object.values( AMF_DATA.providers )[0];
+		toggleUI( provider.supports );
+		return;
+	}
+
+	// Override core styles that allow only two filter inputs
+	addInlineStyle( `
+		.media-modal-content .media-frame select.attachment-filters { width: 150px }
+		.media-modal-content .media-frame #media-attachment-provider-filter + .spinner { float: right; margin: -25px -0px 5px 15px; }
+	` );
+
+	// Create a new MediaLibraryProviderFilter we later will instantiate
+	var MediaLibraryProviderFilter = wp.media.view.AttachmentFilters.extend({
+		id: 'media-attachment-provider-filter',
+
+		createFilters: function() {
+			var filters = {};
+
+			// Formats the 'providers' we've included via wp_localize_script()
+			_.each( AMF_DATA.providers || {}, function( value, index ) {
+				filters[ index ] = {
+					text: value.name,
+					props: {
+						provider: index,
+					}
+				};
+			});
+			this.filters = filters;
+		},
+
+		select: function() {
+			var model = this.model,
+				value = Object.keys( AMF_DATA.providers )[0],
+				props = model.toJSON();
+
+			_.find( this.filters, function( filter, id ) {
+				var equal = _.all( filter.props, function( prop, key ) {
+					return prop === ( _.isUndefined( props[ key ] ) ? null : props[ key ] );
+				});
+
+				if ( equal ) {
+					return value = id;
```

Comment:
Oh lol, I didn't even see that, this is copied from the upstream `AttachmentFilters` "abstract" class almost verbatim. I had to change the logic where `value` is originally set because it'll only match an empty select value by default whereas our default will be whichever provider is first in the list.

Reply:
It does seem to be a shorthand as it's setting the outer value variable. Seems like a really weird use of `_.find()` though. I'm just hesitant to modify it too much from the core version.

Reply:
Yeah, OK, so if this is original code, let's just fix it according to our standards...?

```suggestion
					value = id;
					return value;
```

?